datahub-web-react/src/app/ingest/source/executions/IngestionSourceExecutionList.tsx

```diff
@@ -0,0 +1,163 @@
+    lastRefresh: number;
+    onRefresh: () => void;
+};
+
+export const IngestionSourceExecutionList = ({ urn, isExpanded, lastRefresh, onRefresh }: Props) => {
+    const [focusExecutionUrn, setFocusExecutionUrn] = useState<undefined | string>(undefined);
+
+    const start = 0;
+    const count = 10; // Load 10 items at a time.
+
+    const { loading, data, error, refetch } = useGetIngestionSourceQuery({
+        variables: {
+            urn,
+            runStart: start,
+            runCount: count,
+        },
+    });
+
+    function hasActiveExecution() {
+        return !!data?.ingestionSource?.executions?.executionRequests.find((request) =>
+            isExecutionRequestActive(request as ExecutionRequest),
+        );
+    }
+    useRefreshIngestionData(refetch, hasActiveExecution);
+
+    const [cancelExecutionRequestMutation] = useCancelIngestionExecutionRequestMutation();
+    const [rollbackIngestion] = useRollbackIngestionMutation();
+
+    useEffect(() => {
+        if (isExpanded) {
+            refetch();
+        }
+    }, [lastRefresh, isExpanded, refetch]);
+
+    const handleViewDetails = (focusUrn: string) => {
+        setFocusExecutionUrn(focusUrn);
+    };
+
+    const onCancelExecutionRequest = (executionUrn: string) => {
+        cancelExecutionRequestMutation({
+            variables: {
+                input: {
+                    ingestionSourceUrn: urn,
+                    executionRequestUrn: executionUrn,
+                },
+            },
+        })
+            .catch((e) => {
+                message.destroy();
+                message.error({
+                    content: `Failed to cancel execution!: \n ${e.message || ''}`,
+                    duration: 3,
+                });
+            })
+            .finally(() => {
+                message.success({
+                    content: `Successfully submitted cancellation request!`,
+                    duration: 3,
+                });
+                // Refresh once a job was cancelled.
+                setTimeout(() => onRefresh(), 2000);
+            });
+    };
+
+    const handleCancelExecution = (executionUrn: string) => {
+        Modal.confirm({
+            title: `Confirm Cancel`,
+            content:
+                'Cancelling an running execution will NOT remove any data that has already been ingested. You can use the DataHub CLI to rollback this ingestion run.',
+            onOk() {
+                onCancelExecutionRequest(executionUrn);
+            },
+            onCancel() {},
+            okText: 'Cancel',
+            cancelText: 'Close',
+            maskClosable: true,
+            closable: true,
+        });
+    };
+
+    function handleRollbackExecution(runId: string) {
+        Modal.confirm({
+            title: `Confirm Rollback`,
+            content: (
+                <div>
+                    Rolling back this ingestion run will delete any data specifically associated with with this run. If
```

Comment:
I think one piece of this copy that is a bit confusing is the idea of "overlapping data".

I'd rather just say: Rolling back this ingestion run will remove any new data ingested during the run. This may exclude data that was previously extracted, but did not change during this run.

Reply:
We can do this in a followup